### DISCUSSION TITLE
Is mail feature in v11.28 working with Gmail?

### DISCUSSION THREAD

**Discussion by Ringoinmel** [chamilo/chamilo-lms#6004] [General] [answered]
I put all SMTP settings to \chamilo\app\config\mail.conf.php and try to send mail when I give the comments to the assignment as below.
![image](https://github.com/user-attachments/assets/86a826f1-ab9b-4cf2-bfc6-1eb2b836ba3b)

But the student haven't receive the comments by email.  Here is what the settings I put it.

$platform_email['SMTP_FROM_EMAIL'] = 'summitlearning.au@gmail.com'; //See the function __construct() in main/inc/lib/notification.lib.php for more details on how the SMTP FROM email is defined and what to indicate here if needed to override users configuration
$platform_email['SMTP_FROM_NAME'] = 'Summit Learning';
$platform_email['SMTP_HOST'] = 'smtp.gmail.com'; // If using SMTP use the domain name example: mywebmail.example.net
$platform_email['SMTP_PORT'] = 587;
$platform_email['SMTP_MAILER'] = 'smtp.gmail.com'; // mail, sendmail or smtp (Windows probably only supports smtp)
$platform_email['SMTP_AUTH'] = 1;
$platform_email['SMTP_USER'] = 'summitlearning.au@gmail.com';
$platform_email['SMTP_PASS'] = 'xxxxxxxxxxxxx';
$platform_email['SMTP_CHARSET'] = 'UTF-8';
$platform_email['SMTP_UNIQUE_SENDER'] = 0; // to send all mails from the same user
$platform_email['SMTP_DEBUG'] = 0; // change to 1 to enable smtp debug
$platform_email['SMTP_SECURE'] = 'tls'; // if you're using SSL: ssl; or TLS: tls. (only used if SMTP_AUTH==1)
$platform_email['SMTP_UNIQUE_REPLY_TO'] = 0; // to use AWS SMS service, SMTP_UNIQUE_SENDER and this have to be = 1

**Comment by Ringoinmel** [comment]
Check error log and it said
24/12/28 11:20:16 << 530-5.7.0 Must issue a STARTTLS command first. For more information, go to<EOL>530-5.7.0  https://support.google.com/a/answer/3221692 and review RFC 3207<EOL>530 5.7.0 specifications. d9443c01a7336-219dc9d9927sm140644405ad.134 - gsmtp<EOL>

Any idea?

**Reply by Ringoinmel** [reply]
After research, I think it is missing to enable STARTTLS.  Anyone know how to add it to mail.conf.php

I try $platform_email['SMTP_STARTTLS'] = 1;

but doesn't work

**Comment by Ringoinmel** [comment]
Anyone can help?

**Comment by ywarnier** [comment, marked as answer]
Hi @Ringoinmel 

Sorry for the late answer. I haven't tried gmail for a long time so I cannot guarantee it works *but*:

If you set SMTP_DEBUG = 1 in mail.conf.php (do that at a time of less usage because it does print debug info on screen), you should be able to load the "E-mail tester" screen from the admin panel and test sending a few mails, then seeing what the PHP error log tells you (which will be more detailed).

Also, for Amazon, for example, it is required to have SMTP_UNIQUE_SENDER = 1. Might be useful.

In the configuration you posted above, and despite the comment on the right side that specifically says "mail, sendmail or smtp", you used 'smtp.gmail.com' for $platform_email['SMTP_MAILER'].

This would be my first guess.
By default, Chamilo uses "mail" which is the local mail service (on the Chamilo server itself), which rarely works if this is not the main server for that domain name. Use "smtp" instead. You have probably gone through that initially, but this might have failed in combination with some other setting that was wrong at the time. SMTP_MAILER is the *method*, not the server.

**Reply by Ringoinmel** [reply]
Problem fixed.  The settings should be

$platform_email['SMTP_MAILER'] = 'mail';

**Comment by Coursenligne** [comment]
I just made several tests and occured that I can't send using gmail smtp.
This seems to be due to authentication 2 factors and OAuth2 used by Google.
I used another acount using "normal" password and standard smtp and it worked straight away.
I hope this helps.

**Comment by peninsula3** [comment]
Gmail SMTP in Chamilo v11.28 often fails due to 2FA/OAuth2. Use a standard account or App Password and set $platform_email['SMTP_MAILER'] = 'mail'. Enable $platform_email['SMTP_DEBUG'] = 1 to test.
